src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnsafeRelaxedJavaScriptEncoder.cs

```diff
@@ -225,7 +260,7 @@ public override unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf
                     {
                         if (!_allowedCharacters.IsUnicodeScalarAllowed(ptr[idx]))
                         {
-                            goto Return;
+                            return idx;
```

Comment:
@kunalspathak I made those changes in an attempt to improve overall clarity, however I can't help but think that there might be a legitimate performance reason for using `goto` instead of inlining the return statement.